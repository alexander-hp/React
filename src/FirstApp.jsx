// TODO si es que la variable no va a cambiar de valor
// TODO o esta asociada a algun hook entonces no debe de ir en
// TODO el functionalcomponent, osea no dentro de la function
const newMessage = [1, 2, 3, 4, 5];

// ? Como imprimir un objeto
const message = {
  title: 'Titulo del objeto',
  message: 'mensaje obj',
};

// ? devolvemos un mensaje en una funcion
function getMessage() {
  return 'Mensaje obtenido';
}

const getSuma = (a, b) => {
  return a + b;
};

// ? en el parametro estamos recibiendo las props ya desestructurada
export default function FirstApp({ title }) {
  return (
    // ? esto sustituye al fragment para no anidar codigo a otro div
    // ? el cual podria estropear los estilos que tenemos: <>
    <>
      <h1>Alexander</h1>
      {/* Podemos poner js o variables aqui todo a excepcion de un objeto */}
      <h2> {12 * 15} </h2>
      <h2> {newMessage} </h2>
      <h2> {JSON.stringify(message)} </h2>
      <h2> {getMessage()} </h2>
      <h2> {getSuma(12, 15)} </h2>
      {/* Las Props es el contexto al que se le pasa un componente */}
      <h2> Props: {title} </h2>
      <p>Soy un parrafo</p>
    </>
  );
}
