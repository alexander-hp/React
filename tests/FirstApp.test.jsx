/** @jest-environment jsdom */
import { render } from '@testing-library/react';
import FirstApp from '../src/FirstApp';

describe('Pruebas en <FirstApp>', () => {
  test('debe hacer match con el snapshot', () => {
    // ? estamos renderizando el componente FirstApp
    const title = 'TITULO';
    // ? render nos devolverar todo el nodo de html, como el DOM
    const { container } = render(<FirstApp title={title} />);
    // console.log(container);
    /* 
    ? si ejecutamos lo siguiente, nos creara una carpeta llamada __snapshots__
    ? el cual nos dara todo lo que nosotros retornamos en el div o fragment
    * Esto nos ayuda a saber si algo cambio en nuestro componente desde la ultima
    * vez que realizamos nuestro snapshot, si lo queremos actualizar podemos correr
    * de nuevo el test y precionar u
    */
    expect(container).toMatchSnapshot();
  });

  test('Debe de mostrar un titulo en h2', () => {
    const title = 'TITULO';
    const { container, getByText } = render(<FirstApp title={title} />);

    // ? Aqui solo decimos que Props.title: TITULO debe existir en ese componente
    expect(getByText('Props.title: ' + title)).toBeTruthy();

    // ? Estamos obteniendo un h2 en el DOM
    const h2 = container.querySelector('h2');
    console.log(h2.innerHTML);

    // ? Aqui checamos que deba de contener TITULO n el h2 que obtuvimos
    expect(h2.innerHTML).toContain(title);
  });

  test('Debe de mostrar el titulo por un id', () => {
    const title = 'TITULO';
    const { container, getByText, getByTestId } = render(
      <FirstApp title={title} />
    );

    // ? Checamos que un tag tenga como id test-title, si existe
    expect(getByTestId('test-title')).toBeTruthy();
    // ? que contenga el texto que tenemos en ese id
    expect(getByTestId('test-title').innerHTML).toContain(title);
    // ? Tambien lo podriamos hacer con toBe
  });

  test('Debe de mostrar el subtitle enviado por props', () => {
    const title = 'TITULO';
    const subtitle = 'Soy un SUBTITULO';
    const { container, getAllByText } = render(
      // ?Enviamos los props
      <FirstApp title={title} subtitle={subtitle} />
    );

    console.log(getAllByText(subtitle));

    // ? Checamos cuantos coinciden con ese texto, nos lo retorna en arreglo de htmlElement
    // ? tiene que ser de 2 en este caso
    expect(getAllByText(subtitle).length).toBe(2);
  });
});
