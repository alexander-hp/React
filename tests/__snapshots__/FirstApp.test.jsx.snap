// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp> debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Alexander
  </h1>
  <h2>
    TITULO
  </h2>
  <div
    data-testid="test-title"
  >
     
    TITULO
     
  </div>
  <p>
    asignado desde funcion
  </p>
  <p>
    asignado desde funcion
  </p>
  <h2>
     
    180
     
  </h2>
  <h2>
     
    1
    2
    3
    4
    5
     
  </h2>
  <h2>
     
    {"title":"Titulo del objeto","message":"mensaje obj"}
     
  </h2>
  <h2>
     
    Mensaje obtenido
     
  </h2>
  <h2>
     
    27
     
  </h2>
  <h2
    class="propTitle"
  >
     Props.title: 
    TITULO
     
  </h2>
  <h2>
     Props.subtitle: 
    asignado desde funcion
     
  </h2>
  <h2>
     Props.number: 
    NaN
     
  </h2>
  <h2>
     Props.boolean: 
     
  </h2>
  <p>
    Soy un parrafo
  </p>
</div>
`;
